chunky/src/java/se/llbit/chunky/model/FireModel.java

```diff
@@ -38,10 +40,14 @@
           new Quad(new Vector3(0, 0, 1), new Vector3(1, 0, 0), new Vector3(0, 1, 1),
               new Vector4(1, 0, 0, 1)),};
 
-  public static boolean intersect(Ray ray, AnimatedTexture[] texture) {
+  public static boolean intersect(Ray ray, AnimatedTexture[] texture, double time) {
     boolean hit = false;
     ray.t = Double.POSITIVE_INFINITY;
-    int i = 0xF & (ray.getCurrentData() >> BlockData.LILY_PAD_ROTATION);
+    Vector3 position = new Vector3(ray.o);
+    position.scaleAdd(Ray.OFFSET, ray.d);
+    int i = (0xF & (ray.getCurrentData() >> BlockData.LILY_PAD_ROTATION))
```

Comment:
`i` used to be `[0, 15]`

Reply:
`i` dosen't have to be in the range 0,15. The animation frame used = `i % numFrames`.